x/incentives/keeper/gauge.go

```diff
@@ -266,13 +271,6 @@ func (k Keeper) CreateGroup(ctx sdk.Context, coins sdk.Coins, numEpochPaidOver u
 	k.SetGroup(ctx, newGroup)
 	k.SetLastGaugeID(ctx, gauge.Id)
 
-	// TODO: check if this is necessary.
-	// Tracked in issue https://github.com/osmosis-labs/osmosis/issues/6405
-	combinedKeys := combineKeys(types.KeyPrefixUpcomingGauges, getTimeKey(gauge.StartTime))
-
-	if err := k.CreateGaugeRefKeys(ctx, &gauge, combinedKeys); err != nil {
-		return 0, err
-	}
```

Comment:
Note: the decision on not needing this has been finalized.

The test for `CreateGroup` are to be reviewed and update for final invariants in: https://github.com/osmosis-labs/osmosis/issues/6505